### PR TITLE
Add Team Score.

### PR DESCRIPTION
**Pull request by RLMarvin**
This adds number of goals scored by each team.
Potential fix for #323 & #324

### CODE REVIEW

**Comment by tarehart** [comment]
Awesome, thanks Marvin! I'll approve this as soon as I get home to test it.

**Comment by whatisaphone** [comment]
Awesome, we've been needing this for a while!

What do you think about using an array instead of two separate fields?

**Comment by RLMarvin** [comment]
I think it could be more future-proof, as in if we add team specific info it'll have structure already in place.
However it adds unnecessary complexity if it's an array that'll only ever hold one value.

What's your thoughts?

**Comment by tarehart** [comment]
You both have good points. I feel like there are two separate decisions here:
- Do we want to start a struct for team specific info
  - Pleasantly extensible, as noted above
  - Mildly more complex
- Do we want to express the data as an array or two specific fields
  - An array would allow people to index into it more elegantly with their bot's team number. Some people would probably also consider an array to be better aesthetically.

**Comment by RLMarvin** [comment]
Thanks for the clarification!
I'm fine with doing both, but that may delay the feature a bit.

**Comment by tarehart** [comment]
If you think the answer to both is 'yes', I'm happy to pitch in.

**Comment by whatisaphone** [comment]
I skimmed through the `ATeam*` fields and I don't think there's anything else in there besides score that can change in the middle of a game. But there's lots of mildly interesting data that should be constant during a game. The ones that jump out at me are TeamName/TeamColor (useful for UIs or tournament runners?), ForfeitVote (let bots forfeit!), TemporarySpawnSpots and/or SecondaryStarts (spawn positions after a demo). If you can see a future where we add any of those, let's do a struct, otherwise not.

I like your idea of moving it up to `GameTickPacket`.

**Comment by RLMarvin** [comment]
Thanks for the feedback! I'll try making the changes, hopefully this goes out before ep6!

File: src/main/cpp/RLBotInterface/RLBotMessages/MessageTranslation/FlatbufferTranslator.cpp [hunk 1]
```diff
@@ -156,6 +164,14 @@ namespace FlatbufferTranslator {
 			}
 		}
 
+		auto teams = flatPacket->teams();
+		if (teams) {
+			packet->NumTeams = teams->size();
+			for (int i = 0; i < teams->size() && i < CONST_MaxTeams; i++) {
+			fillTeamStruct(teams->Get(i), &packet->Teams[i]);
```
Comment (DomNomNom) [inline src/main/cpp/RLBotInterface/RLBotMessages/MessageTranslation/FlatbufferTranslator.cpp]: indentation

File: src/main/cpp/RLBotInterface/RLBotMessages/MessageTranslation/FlatbufferTranslator.cpp [hunk 2]
```diff
@@ -156,6 +164,14 @@ namespace FlatbufferTranslator {
 			}
 		}
 
+		auto teams = flatPacket->teams();
+		if (teams) {
+			packet->NumTeams = teams->size();
+			for (int i = 0; i < teams->size() && i < CONST_MaxTeams; i++) {
```
Comment (DomNomNom) [inline src/main/cpp/RLBotInterface/RLBotMessages/MessageTranslation/FlatbufferTranslator.cpp]:
```suggestion
			packet->NumTeams = min(teams->size(), CONST_MaxTeams);
			for (int i = 0; i < packet->NumTeams; i++) {
```

File: src/main/python/rlbot/utils/structures/game_data_struct.py [hunk 3]
```diff
@@ -145,34 +154,52 @@ class ByteBuffer(Struct):
 # This negates all x and y values for balls and cars and rotates yaw 180 degrees.
 def rotate_game_tick_packet_boost_omitted(game_tick_packet):
     # Negate all x,y values for ball
-    game_tick_packet.game_ball.physics.location.x = -1 * game_tick_packet.game_ball.physics.location.x
-    game_tick_packet.game_ball.physics.location.y = -1 * game_tick_packet.game_ball.physics.location.y
-    game_tick_packet.game_ball.physics.velocity.x = -1 * game_tick_packet.game_ball.physics.velocity.x
-    game_tick_packet.game_ball.physics.velocity.y = -1 * game_tick_packet.game_ball.physics.velocity.y
+    game_tick_packet.game_ball.physics.location.x = - \
+        1 * game_tick_packet.game_ball.physics.location.x
```
Comment (DomNomNom) [inline src/main/python/rlbot/utils/structures/game_data_struct.py]: The  `-` being on a different line to the `1` makes it easy to mis-read by missing the negation.
Put the `-` on the new line too.

Comment (RLMarvin) [reply src/main/python/rlbot/utils/structures/game_data_struct.py]: This was the fault of my auto-formatted :(

**Pull request merged**